inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/KafkaSink.java

```diff
@@ -0,0 +1,445 @@
+import static org.apache.inlong.agent.constant.AgentConstants.DEFAULT_KAFKA_SINK_SEND_KEY_SERIALIZER;
+import static org.apache.inlong.agent.constant.AgentConstants.DEFAULT_KAFKA_SINK_SEND_VALUE_SERIALIZER;
+import static org.apache.inlong.agent.constant.AgentConstants.DEFAULT_KAFKA_SINK_SYNC_SEND_TIMEOUT_MS;
+import static org.apache.inlong.agent.constant.AgentConstants.DEFAULT_PRODUCER_NUM;
+import static org.apache.inlong.agent.constant.AgentConstants.DEFAULT_SEND_QUEUE_SIZE;
+import static org.apache.inlong.agent.constant.AgentConstants.KAFKA_PRODUCER_ENABLE_ASYNC_SEND;
+import static org.apache.inlong.agent.constant.AgentConstants.KAFKA_SINK_PRODUCER_NUM;
+import static org.apache.inlong.agent.constant.AgentConstants.KAFKA_SINK_SEND_QUEUE_SIZE;
+import static org.apache.kafka.clients.CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_JAAS_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_MECHANISM;
+
+public class KafkaSink extends AbstractSink {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaSink.class);
+    private static final ExecutorService EXECUTOR_SERVICE = new ThreadPoolExecutor(0, Integer.MAX_VALUE,
+            60L, TimeUnit.SECONDS, new SynchronousQueue<>(), new AgentThreadFactory("KafkaSink"));
+    private final AgentConfiguration agentConf = AgentConfiguration.getAgentConf();
+    private TaskPositionManager taskPositionManager;
+    private volatile boolean shutdown = false;
+
+    private List<MQClusterInfo> mqClusterInfos;
+    private String topic;
+    private List<KafkaSender> kafkaSenders;
+    private static final AtomicInteger KAFKA_SENDER_INDEX = new AtomicInteger(0);
+
+    private LinkedBlockingQueue<BatchProxyMessage> kafkaSendQueue;
+
+    private int producerNum;
+    private boolean asyncSend;
+
+    @Override
+    public void init(JobProfile jobConf) {
+        super.init(jobConf);
+        taskPositionManager = TaskPositionManager.getInstance();
+        int sendQueueSize = agentConf.getInt(KAFKA_SINK_SEND_QUEUE_SIZE, DEFAULT_SEND_QUEUE_SIZE);
+        kafkaSendQueue = new LinkedBlockingQueue<>(sendQueueSize);
+        producerNum = agentConf.getInt(KAFKA_SINK_PRODUCER_NUM, DEFAULT_PRODUCER_NUM);
+        asyncSend = agentConf.getBoolean(KAFKA_PRODUCER_ENABLE_ASYNC_SEND, DEFAULT_ENABLE_ASYNC_SEND);
+
+        mqClusterInfos = jobConf.getMqClusters();
+        Preconditions.checkArgument(ObjectUtils.isNotEmpty(jobConf.getMqTopic()) && jobConf.getMqTopic().isValid(),
+                "no valid kafka topic config");
+        topic = jobConf.getMqTopic().getTopic();
+
+        kafkaSenders = new ArrayList<>();
+        initKafkaSender();
+        EXECUTOR_SERVICE.execute(sendDataThread());
+        EXECUTOR_SERVICE.execute(flushCache());
+    }
+
+    @Override
+    public void write(Message message) {
+        try {
+            if (message != null) {
```

Comment:
The nesting of this code is too deep, it is recommended to use guard statements to simplify.

For example:
```java
if (message == null) {
    return;
}

// other process in else
```

Reply:
Okay, thx. I'll optimize it.